File: src/main/java/Ex10_Loops.java

public class Ex10_Loops {
    public static void main(String[] args) {
        fruitsPrinter("Banana", "Apple", "Orange");
    }

    static void fruitsPrinter(
        String... fruits
    ) {
        for (int i = 0; i < fruits.length; i++) {
            System.out.println(i + " " + fruits[i]);
        }
    }
}

File: src/main/kotlin/Ex10_Loops.kt
fun main() {
  arrayOf("Apple", "Banana", "Orange").printer()
}

fun Array<String>.printer() {
  for (i in 0 .. size - 1) {
    val fruit: String = this[i]
    println("$i: $fruit")
  }

  for (i in 0 until  size) {
    val fruit: String = this[i]
    // println("$i: $fruit")
  }

  for (i in 0 until  lastIndex) {
    val fruit: String = this[i]
    // println("$i: $fruit")
  }

  for (i in indices) {
    val fruit: String = this[i]
    // println("$i: $fruit")
  }
}